Src/Models/Model.cpp

```diff
@@ -37,25 +37,31 @@ Model::Model(Assimp::Importer* importer, GraphicsResources* gr, const PGE::Strin
         aiProcess_SortByPType |
         aiProcess_FindDegenerates
     );
+    thisScene = scene;
 
     PGE::String err = importer->GetErrorString();
     PGE_ASSERT(err.isEmpty(), "Failed to load model (err: " + err + ")");
-
     materialCount = scene->mNumMaterials;
+
+    //printf("Loading model: %s. Material count: %i\n", filename.cstr(), materialCount);
+
     materials = new PGE::Material*[materialCount];
     for (unsigned int i = 0; i < materialCount; i++) {
         aiString texturePath;
         PGE_ASSERT(scene->mMaterials[i]->GetTexture(aiTextureType_DIFFUSE, 0, &texturePath) == aiReturn_SUCCESS, "Texture for model " + filename + " failed to load.");
         PGE::String textureName = PGE::String(texturePath.C_Str()).replace("\\", "/");
         int lastSlash = textureName.findLast("/").getPosition();
         textureName = textureName.substr(lastSlash + 1, textureName.length() - lastSlash - 5);
+        //printf("Resolved to Texture: %s\n", textureName.cstr());
         materials[i] = new PGE::Material(shader, gr->getTexture(path + textureName));
     }
 
     meshCount = scene->mNumMeshes;
     meshes = new PGE::Mesh*[meshCount];
+    originMaterials = new unsigned int[meshCount];
```

Comment:
You have a memory leak here. This array is not deleted at any point.

Reply:
Fixed.
This array is used to store pointers to the original material to when a model is not given a custom material.